File: front/src/app/models/planet.ts
import { Moon } from "./moon";
import {FormControl, FormGroup, Validators} from '@angular/forms';

export class Planet
{
	constructor(public id: number,
				public name: string,
				public diameter: number,
				public distance_from_sun: number,
				public description: string,
				public image: string,
				public moons: Array<Moon>) {
	}

	static asFormGroup(planet: Planet): FormGroup
	{
		return new FormGroup({
			id: new FormControl(planet.id, Validators.required),
			name: new FormControl(planet.name, Validators.required),
			diameter: new FormControl(planet.diameter, Validators.required),
			distance_from_sun: new FormControl(planet.distance_from_sun, Validators.required),
			description: new FormControl(planet.description, Validators.required),
		})
	};
}

File: backend/src/main/java/fr/redship/projectwebapidb/model/Planet.java
package fr.redship.projectwebapidb.model;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Table(name="planets")
@Entity
@Getter @Setter
@JsonIdentityInfo(
        generator = ObjectIdGenerators.PropertyGenerator.class,
        property = "id"
)
public class Planet {

    @Id
    private Long id;

    private String name;
    private double diameter;
    private double distance_from_sun;
    private String description;

    @OneToMany(mappedBy = "planet", cascade = CascadeType.ALL)
    private List<Moon> moons;

}
